sentry-core/src/transport.rs

```diff
@@ -14,15 +14,19 @@ pub trait Transport: Send + Sync + 'static {
     /// [`Envelope`]: struct.Envelope.html
     fn send_envelope(&self, envelope: Envelope);
 
-    /// Drains the queue if there is one.
+    /// Flushes the transport queue if there is one.
     ///
-    /// The default implementation does nothing.  If the queue was successfully
-    /// shutdowned the return value should be `true` or `false` if events were
-    /// left in it.
-    fn shutdown(&self, timeout: Duration) -> bool {
+    /// If the queue was successfully drained, the return value should be
+    /// `true` or `false` if events were left in it.
+    fn flush(&self, timeout: Duration) -> bool {
```

Comment:
As we're re-writing this I can't help but wonder if Duration is really that great here.  Maybe something that can be dropped instead with a wrapper that can drop on a Duration.  Admittedly that's probably a bit heavy weight and overthinking it.  I have no usecase :)

Reply:
I don’t quite understand what you mean?

Reply:
Never mind, I'm not sure this random musing is very useful.  I'm struggling to come up with a reasonable use-case for something else that a Duration.